xbmc/filesystem/SMBFile.cpp

```diff
@@ -100,26 +100,23 @@ void CSMB::Init()
     std::string home(getenv("HOME"));
     URIUtils::RemoveSlashAtEnd(home);
     snprintf(smb_conf, sizeof(smb_conf), "%s/.smb", home.c_str());
```

Comment:
not specific to this PR really but it would be nice to get rid of the char buffer on the stack and use a string, maybe with `StringUtils::Format` or just concats. snprintf on the stack should probably be avoided.

Not a blocking issue, just a note.